e2etest/e2etest_suite_test.go

```diff
@@ -70,3 +83,218 @@ func TestE2E(t *testing.T) {
 	gomega.RegisterFailHandler(framework.Fail)
 	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "E2E Suite", r)
 }
+
+var _ = ginkgo.BeforeSuite(func() {
+	cs, err := framework.LoadClientset()
+	framework.ExpectNoError(err)
+	_, err = cs.CoreV1().ConfigMaps(testNameSpace).Get(context.TODO(), "config", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		err = configureMetalLB(cs)
+		framework.ExpectNoError(err)
+	} else if err == nil {
+		metallbCMExisted = true
+	} else {
+		framework.ExpectNoError(err)
+	}
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	if !metallbCMExisted {
+		cs, err := framework.LoadClientset()
+		framework.ExpectNoError(err)
+		err = deleteMetalLBConfiguration(cs)
+		framework.ExpectNoError(err)
+	}
+})
+
+// Run a BGP router in a container for all of the speakers to peer with.
+func configureFRR(cs clientset.Interface) error {
+	// We need the ip of each node in the cluster to place them in the BGP router configuration file
+	// Each node will peer with this router.
+	nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+
+	// TODO: The IP address handling will need updates to add support for IPv6.
+	nodesIPs := make([]string, 0, len(nodes.Items))
+	for _, node := range nodes.Items {
+		for i := range node.Status.Addresses {
+			if node.Status.Addresses[i].Type == "InternalIP" {
+				nodesIPs = append(nodesIPs, node.Status.Addresses[i].Address)
+			}
+		}
+	}
+
+	if len(nodesIPs) != 3 {
+		return fmt.Errorf("Expected 3 nodes, got %d", len(nodesIPs))
+	}
+
+	// Create temp dir to update the BGP router configuration file so that we can include the current nodes IPs.
+	frrConfigTestDir, err = ioutil.TempDir(".", "frr-conf")
+	if err != nil {
+		return fmt.Errorf("Failed to create temp FRR config directory: %s", err)
+	}
+
+	err = CopyDir(consts.BGPTestFRRConfigDir, frrConfigTestDir)
+	if err != nil {
+		return fmt.Errorf("Failed to copy FRR config directory: %s", err)
+	}
+
+	bgpdConfigTestFile := fmt.Sprintf("%s/%s", frrConfigTestDir, consts.BGPDConfigFileName)
```

Comment:
Should this be created in `consts.BGPTestFRRConfigDir`?

Reply:
the frrConfigTestDir is a temp dir getting a random name starting with the prefix "frr-conf":
`frrConfigTestDir, err = ioutil.TempDir(".", "frr-conf")`